src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+                    case SyntaxKind.Argument:
+                        return TryReplaceWithLocalFunction(source, parent, out documentUpdateInfo);
+                    default:
+                        documentUpdateInfo = default;
+                        return false;
+                }
+            }
+
+            private bool TryConvertIfInMemberAccessExpression(
+               QueryExpressionSyntax source,
+               MemberAccessExpressionSyntax memberAccessExpression,
+               out DocumentUpdateInfo documentUpdateInfo)
+            {
+                if (memberAccessExpression.Parent is InvocationExpressionSyntax invocationExpression)
+                {
+                    // This also covers generic names (i.e. with type arguments) like 'ToList<int>'. 
+                    // The ValueText is still just 'ToList'. 
+                    switch (memberAccessExpression.Name.Identifier.ValueText)
+                    {
+                        case nameof(Enumerable.ToList):
+                            return TryConvertIfInToListInvocation(source, invocationExpression, out documentUpdateInfo);
+                        case nameof(Enumerable.Count):
+                            return TryConvertIfInCountInvocation(source, invocationExpression, out documentUpdateInfo);
+                        default:
+                            return TryReplaceWithLocalFunction(source, memberAccessExpression, out documentUpdateInfo);
+                    }
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInCountInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                if (_semanticModel.GetSymbolInfo(invocationExpression, _cancellationToken).Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 0 && methodSymbol.ReturnType?.SpecialType == SpecialType.System_Int32)
+                {
+                    // before var count = (from a in b select a).Count();
+                    // after
+                    // var count = 0;
+                    // foreach (var a in b)
+                    // {
+                    //     count++;
+                    //  }
+                    return TryConvertIfInInvocation(
+                            source,
+                            invocationExpression,
+                            (variableIdentifier, expression) => SyntaxFactory.ExpressionStatement(
+                                SyntaxFactory.PostfixUnaryExpression(SyntaxKind.PostIncrementExpression, variableIdentifier)), // Generating 'count++'
+                            SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, SyntaxFactory.Literal(0)), // count = 0
+                            variableName: "count",
```

Comment:
>count [](start = 43, length = 5)

Are we sure the name is not going to conflict with anything in scope (local, parameter, field, property, method, ... ) ?

Reply:
yes, we use 'count' as a prefix and generate a unique name. There are tests covering this.

---
In reply to: [176929177](https://github.com/dotnet/roslyn/pull/25362#discussion_r176929177) [](ancestors = 176929177)

Reply:
> yes, we use 'count' as a prefix and generate a unique name. There are tests covering this.

Do we have any test where ```count``` is used in a nested scope?

---
In reply to: [177578926](https://github.com/dotnet/roslyn/pull/25362#discussion_r177578926) [](ancestors = 177578926,176929177)

Reply:
For example, in a block before or after the query.

---
In reply to: [177580584](https://github.com/dotnet/roslyn/pull/25362#discussion_r177580584) [](ancestors = 177580584,177578926,176929177)

Reply:
ReturnCountNameGeneration

---
In reply to: [177580584](https://github.com/dotnet/roslyn/pull/25362#discussion_r177580584) [](ancestors = 177580584,177578926,176929177)

Reply:
let me try to extend for neighbor block

---
In reply to: [177587632](https://github.com/dotnet/roslyn/pull/25362#discussion_r177587632) [](ancestors = 177587632,177580584,177578926,176929177)

Reply:
> ReturnCountNameGeneration

This is not about use in nested scope

---
In reply to: [177587753](https://github.com/dotnet/roslyn/pull/25362#discussion_r177587753) [](ancestors = 177587753,177587632,177580584,177578926,176929177)

Reply:
I've added tests. had to switch to _semanticFacts.GenerateUniqueName to pass them. Thanks!

---
In reply to: [177587753](https://github.com/dotnet/roslyn/pull/25362#discussion_r177587753) [](ancestors = 177587753,177587632,177580584,177578926,176929177)

Reply:
Here is a test:
using System.Collections.Generic;
using System.Linq;
class C
{
    void N(int c) {}
    void M(IEnumerable<int> nums)
    {
        N(([|from int n1 in nums 
                 from int n2 in nums
                 select n1|]).Count());

        if (true)
        {
            int count = 1;
        }
    }
}

---
In reply to: [177591501](https://github.com/dotnet/roslyn/pull/25362#discussion_r177591501) [](ancestors = 177591501,177587753,177587632,177580584,177578926,176929177)